grayskull/pypi/pypi.py

```diff
@@ -552,11 +567,22 @@ def _get_metadata(self) -> dict:
         version = ""
         if self["package"]["version"].values:
             version = self.get_var_content(self["package"]["version"].values[0])
-        pypi_metadata = self._get_pypi_metadata(name, version)
-        sdist_metadata = self._get_sdist_metadata(
-            sdist_url=pypi_metadata["sdist_url"], name=name
-        )
-        metadata = PyPi._merge_pypi_sdist_metadata(pypi_metadata, sdist_metadata)
+
+        if name.startswith(("http://", "https://")):
+            sdist_url = self._generate_git_archive_tarball_url(name)
+            print(f"This is the sdist_url: {sdist_url}")
+            name = name.split("/")[-1]
+            print(f"This is the package name as extracted from url: {name}")
+            sdist_metadata = self._get_sdist_metadata(sdist_url = sdist_url, name = name)
+            pypi_metadata = {}
+            metadata = PyPi._merge_pypi_sdist_metadata(pypi_metadata, sdist_metadata)
+        else:
+            pypi_metadata = self._get_pypi_metadata(name, version)
+            sdist_metadata = self._get_sdist_metadata(
+                sdist_url=pypi_metadata["sdist_url"], name=name
+            )
+            metadata = PyPi._merge_pypi_sdist_metadata(pypi_metadata, sdist_metadata)
```

Comment:
This is the same line as 578, so we can remove that one and indent this one out for less duplication 👌